Documentation/platform/troubleshooting.md

```diff
@@ -295,3 +295,34 @@ spec:
     - regex: prometheus_replica
       action: LabelDrop
 ```
+
+### High CPU usage by the Prometheus Operator
+
+There cenarios that can cause a high cpu usage by the Prometheus Operator. For instance with the metrics bellow we can get the rate of reconciliations:
+
+```shell
+(sum by(controller,triggered_by) (rate(prometheus_operator_triggered_total[5m]))
+sum by(controller) (rate(prometheus_operator_reconcile_operations_total[5m])))
+```
+
+If this show as being `triggered_by="Secret"`, a solution is to limit the operator to watch only secrets with matching labels using the `--secret-field-selector` argument. Also can use the namespace selectors to limit the amount of namespaces watched by the operator.
+
+There are other cases reported cause by high amount of external services discovered with `additionalScrapeConfigs`. To check the activity of service discovery and target updates you can use the following metrics:
```

Comment:
this applies to Prometheus rather than the operator?

Reply:
You right, removed all prometheus info